exporter/awsxrayexporter/awsxray.go

```diff
@@ -137,3 +151,37 @@ func wrapErrorIfBadRequest(err error) error {
 	}
 	return err
 }
+
+// encodeOtlpAsBase64 builds bytes from traces and generate base64 value for them
+func encodeOtlpAsBase64(td ptrace.Traces, cfg *Config) ([]*string, error) {
+	var documents []*string
+	for i := 0; i < td.ResourceSpans().Len(); i++ {
+		// 1. build a new trace with one resource span
+		singleTrace := ptrace.NewTraces()
+		td.ResourceSpans().At(i).CopyTo(singleTrace.ResourceSpans().AppendEmpty())
+
+		// 2. append index configuration to resource span as attributes, such that X-Ray Service build indexes based on them.
+		injectIndexConfigIntoOtlpPayload(singleTrace.ResourceSpans().At(0), cfg)
+
+		// 3. Marshal single trace into proto bytes
+		bytes, err := ptraceotlp.NewExportRequestFromTraces(singleTrace).MarshalProto()
+		if err != nil {
+			return nil, fmt.Errorf("failed to marshal traces: %w", err)
+		}
+
+		// 4. build bytes into base64 and append with PROTOCOL HEADER at the beginning
+		base64Str := otlpFormatPrefix + base64.StdEncoding.EncodeToString(bytes)
+		documents = append(documents, &base64Str)
+	}
+
+	return documents, nil
+}
+
+func injectIndexConfigIntoOtlpPayload(resourceSpan ptrace.ResourceSpans, cfg *Config) {
```

Comment:
This will lose the `awsxray.AWSXraySegmentAnnotationsAttribute` indexed keys. https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/17855/files#diff-f27fafb1d52fb95e6a499040354bf252b60a816483feac3390a460485f83fef4R349

Need to add them to the indexed attributes. Also, if `IndexAllAttributes` is set to true, do you still need the `IndexedAttributes` slice?

Reply:
> awsxray.AWSXraySegmentAnnotationsAttribute

This index key is embedded in each span, so it is not efficient to get it out in the logic because the logic does not unpack each span. X-Ray Service will be a better place to handle that. 

>  Also, if IndexAllAttributes is set to true, do you still need the IndexedAttributes slice? 

Ideally we don't need that, but the idea to propagate both of them and let X-Ray Service to make the decision. So i think it's okay to keep both of them.